src/scores/continuous/consistent_impl.py

```diff
@@ -0,0 +1,262 @@
+            forecast and observed dimensions must match precisely. Only one of `reduce_dims`
+            and `preserve_dims` can be supplied. The default behaviour if neither are supplied
+            is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom)
+
+    Returns:
+        array of (mean) scores that is consistent for alpha-expectile functional,
+        with the dimensions specified by `dims`. If `dims` is `None`, the returned DataArray will have
+        only one entry, the overall mean score.
+
+    Raises:
+        ValueError: if `alpha` is not strictly between 0 and 1.
+
+    Note:
+        .. math::
+
+            S(x, y) =
+            \\begin{cases}
+            (1 - \\alpha)(\\phi(y) - \\phi(x) - \\phi'(x)(y-x)), & y < x \\\\
+            \\alpha(\\phi(y) - \\phi(x) - \\phi'(x)(y-x)), & x \\leq y
+            \\end{cases}
+
+        where
+
+            - :math:`x` is the forecast
+            - :math:`y` is the observation
+            - :math:`\\alpha` is the expectile level
+            - :math:`\\phi` is a convex function of a single variable
+            - :math:`\\phi'` is the subderivative of :math:`\\phi`
+            - :math:`S(x,y)` is the score.
+
+        Note that if :math:`\\phi` is differentiable then `\\phi'` is its derivative.
+
+    References:
+        -   Gneiting, T. (2011). Making and Evaluating Point Forecasts. Journal of the 
+            American Statistical Association, 106(494), 746–762. 
+            https://doi.org/10.1198/jasa.2011.r10138
+        -   Taggart, R. (2021). Evaluation of point forecasts for extreme events using 
+            consistent scoring functions. Quarterly Journal of the Royal Meteorological 
+            Society, 148(742), 306–320. https://doi.org/10.1002/qj.4206
+
+    """
+    check_alpha(alpha)
+
+    if preserve_dims or reduce_dims:
+        reduce_dims = gather_dimensions(fcst.dims, obs.dims, reduce_dims=reduce_dims, preserve_dims=preserve_dims)
+
+    score_overfcst = (1 - alpha) * (phi(obs) - phi(fcst) - phi_prime(fcst) * (obs - fcst))
+    score_underfcst = alpha * (phi(obs) - phi(fcst) - phi_prime(fcst) * (obs - fcst))
+    result = score_overfcst.where(obs < fcst, score_underfcst)
+    result = apply_weights(result, weights=weights)
+    result = result.mean(dim=reduce_dims)
+
+    return result
+
+
+def consistent_huber_score(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    huber_param: float,
+    phi: Callable[[xr.DataArray], xr.DataArray],
+    phi_prime: Callable[[xr.DataArray], xr.DataArray],
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Score that is consistent for the Huber mean functional with tuning parameter `tuning_param`,
```

Comment:
Is the score a consistent version of the huber score, or just some yet-to-be-explained score. Perhaps, "An adjusted Huber score which has been made consistent according to the Huber mean functional..."?